pkg/apis/thirdparty/kpack/build/v1alpha1/groupversion_info.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2019 the original author or authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package v1 contains API Schema definitions for the kpack v1alpha1 API group
+//
+// This API group is a forked subset of https://github.com/pivotal/kpack/tree/master/pkg/apis/build/v1alpha1
```

Comment:
I know this was a problem for knative since the dependencies there were very large. Is that true with kpack as well?

Reply:
Yeah, did the same for keda, but not sure I understand the rationale, either wrt size or version:
- size: only code actually used is linked
- versions: if our dependency depends on a different version of say a k8s API, that itself is versioned in a different package (assuming compatibility semantics post-alpha, agreed).

Reply:
The primary reason for this is to avoid version hell latter on, not just with system, but also with anything that consumes system (like the cli, or other tools). There were many times where upgrading Knative was difficult not because of API changes, but because of the dependency graph. In once case, we ended up with a version deadlock because Knative Serving and Knative Build had different dependencies.

By importing the types into our project we have a clean dependency graph that is centrally managed and can be tested. Since we're only including the structs that represent the public API, these are unlikely to change in significant ways absent an API version bump (of course alphas will be alphas).

When we introduce a dependency like Keda or kpack, we're not linking against it's runtime, we're consuming it's CRD types. If these projects (or us) were not using go, we'd have to take the same action of creating structs for the types. I'm avoiding the artificial shortcut that incidentally adds massive dependency graph complexity.

Like it or not, the Kubernetes go APIs are not perfectly backwards compatible. They do make breaking changes. We should pick the version of the Kubernetes APIs we want to consume.

Reply:
If you need third party validation of this approach, consider [Knative and Istio](https://github.com/knative/pkg/tree/master/apis/istio/v1alpha3)